optaplanner-examples/src/main/java/org/optaplanner/examples/conferencescheduling/persistence/ConferenceSchedulingGenerator.java

```diff
@@ -348,54 +359,51 @@ private void createSpeakerList(ConferenceSolution solution, int speakerListSize)
             Set<Timeslot> unavailableTimeslotSet;
+            Set<String> preferredTimeslotTagSet = new LinkedHashSet<>();
+            Set<String> undesiredTimeslotTagSet = new LinkedHashSet<>();
             List<Timeslot> timeslotList = solution.getTimeslotList();
             if (random.nextDouble() < 0.10) {
                 double segmentRandom = random.nextDouble();
-                if (segmentRandom < 0.25) {
+                if (segmentRandom < 0.10) {
                     // No mornings
                     unavailableTimeslotSet = timeslotList.stream()
-                            .filter(timeslot -> timeslot.getStartDateTime().toLocalTime().isBefore(LocalTime.of(12,0)))
+                            .filter(timeslot -> timeslot.getStartDateTime().toLocalTime().isBefore(LocalTime.of(12, 0)))
                             .collect(Collectors.toCollection(LinkedHashSet::new));
-                } else if (segmentRandom < 0.50) {
+                    preferredTimeslotTagSet.add(TIMESLOT_AFTER_LUNCH_TAG);
+                } else if (segmentRandom < 0.20) {
                     // No afternoons
                     unavailableTimeslotSet = timeslotList.stream()
-                            .filter(timeslot -> !timeslot.getStartDateTime().toLocalTime().isBefore(LocalTime.of(12,0)))
+                            .filter(timeslot -> !timeslot.getStartDateTime().toLocalTime().isBefore(LocalTime.of(12, 0)))
                             .collect(Collectors.toCollection(LinkedHashSet::new));
-                } else if (segmentRandom < 0.75) {
+                } else if (segmentRandom < 0.30) {
                     // Only 1 day available
                     LocalDate availableDate = timeslotList.get(random.nextInt(timeslotList.size())).getDate();
                     unavailableTimeslotSet = timeslotList.stream()
                             .filter(timeslot -> !timeslot.getDate().equals(availableDate))
                             .collect(Collectors.toCollection(LinkedHashSet::new));
                 } else {
                     unavailableTimeslotSet = timeslotList.stream()
-                            .filter(timeslot -> random.nextDouble() < 0.10)
+                            .filter(timeslot -> random.nextDouble() < 0.05)
                             .collect(Collectors.toCollection(LinkedHashSet::new));
+                    undesiredTimeslotTagSet.add(TIMESLOT_AFTER_LUNCH_TAG);
                 }
             } else {
                 unavailableTimeslotSet = new LinkedHashSet<>(timeslotList.size());
             }
             speaker.setUnavailableTimeslotSet(unavailableTimeslotSet);
-            speaker.setRequiredTimeslotTagSet(new LinkedHashSet<>());
-            speaker.setPreferredTimeslotTagSet(new LinkedHashSet<>());
+            speaker.setRequiredTimeslotTagSet(new LinkedHashSet<>()); // choosing random required/prohibited tags might result in an infeasible solution
```

Comment:
This comment is a todo I presume? Add a TODO in the beginning?

Reply:
This was an explanation why I kept them as an empty set, but actually it's better to keep it as TODO